manifests/init.pp

```diff
@@ -121,6 +144,59 @@
     fail('set_votequorum is true, but no quorum_members have been passed.')
   }
 
+  if $packages {
+    warning('$corosync::packages is deprecated, use $corosync::package_{corosync,pacemaker} instead!')
+
+    package{ $packages:
+      ensure => present,
+    }
+
+    # Ensure no options conflicting with $packages are set:
+
+    if $package_corosync {
+      fail('$corosync::package_corosync and $corosync::packages must not be mixed!')
+    }
+    if $package_pacemaker {
+      fail('$corosync::package_pacemaker and $corosync::packages must not be mixed!')
+    }
+    if $version_corosync {
+      fail('$corosync::version_corosync and $corosync::packages must not be mixed!')
+    }
+    if $version_pacemaker {
+      fail('$corosync::version_pacemaker and $corosync::packages must not be mixed!')
+    }
+  }
+    else {
+      # Handle defaults for new-style package parameters here to allow co-existence with $packages.
+      if ! $package_corosync {
+        $_package_corosync = true
+      }
+
+      if ! $package_pacemaker {
+        $_package_pacemaker = true
+      }
```

Comment:
I consider it a hack to get around the limitations of a declarative language rather than a nice pattern, but I guess both descriptions fit :-)

As for its behaviour: There's indeed a problem here, now that I look at it again. I meant `$package_pacemaker` to be handled correctly, but now that I look at it again I appear to have forgotten the else branch for all four variables in the heat of crafting that pattern. I'll do that on Monday, along with the extra tests.